packages/beacon-node/src/chain/seenCache/seenGossipBlockInput.ts

```diff
@@ -0,0 +1,150 @@
+import {toHexString} from "@chainsafe/ssz";
+import {deneb, RootHex, ssz} from "@lodestar/types";
+import {ChainForkConfig} from "@lodestar/config";
+import {pruneSetToMax} from "@lodestar/utils";
+
+import {
+  BlockInputCacheType,
+  GossipedBlockInput,
+  BlockInput,
+  GossipedInputType,
+  getBlockInput,
+  BlockSource,
+  BlockInputBlobs,
+  BlobsCache,
+} from "../blocks/types.js";
+
+const MAX_GOSSIPINPUT_CACHE = 5;
+// ssz.deneb.BlobSidecars.elementType.fixedSize, 131256 is size for mainnet preset;
+const BLOBSIDECAR_FIXED_SIZE = ssz.deneb.BlobSidecars.elementType.fixedSize ?? 131256;
+
+export class SeenGossipBlockInput {
+  blockInputCache = new Map<RootHex, BlockInputCacheType>();
```

Comment:
this should be `private`?

Reply:
also it's more readable to define `BlockInputCacheType` in this class if it's only used here, cc @wemeetagain if we agree on this for other PRs same to https://github.com/ChainSafe/lodestar/pull/6176#discussion_r1427275332